### ISSUE TITLE
false positive w/unidiomatic-typecheck when comparing two types

### ISSUE THREAD

**Issue by vapier** [PyCQA/pylint#2215]
### Steps to reproduce

Since `unidiomatic-typecheck` was added in the 1.4.x series, and up through 1.9.x, and current 2.0-dev, it incorrectly fires when comparing two `type()` results

example code:
```python
#!/usr/bin/python
d1 = {}
d2 = {}
if type(d1) == type(d2):
    pass
```

this often shows up in `__eq__` of custom classes.

we could write a more convoluted version like:
```python
if isinstance(d1, type(d2)):
    pass
```

but that isn't an improvement over the more straightforward type comparison.

### Current behavior
```sh
$ pylint -r n test.py 
************* Module test
C:  4, 3: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
```

### Expected behavior
there should be no warning when both the LHS and RHS are `type()`.

### pylint --version output
```
pylint 1.9.2, 
astroid 1.6.5
Python 3.6.4 (default, Jan 28 2018, 15:24:24) 
[GCC 6.4.0]
```

cc @cvrebert

**Comment by sushobhit27** [comment]
```python
>>> class Base(object):
...     pass
...
>>> class Derived(Base):
...     pass
...
>>> b = Base()
>>> d = Derived()
>>> type(d) == type(b)
False
>>> isinstance(d, type(b))
True
```
Still the more convoluted version is better.

**Comment by PCManticore** [comment]
This is working as intended. You'd be surprised how many people do `type(value) == type([])`. Changing the check to look only for one type call will reduce the amount of occurrences it is capable of finding.

**Comment by vapier** [comment]
fair enough
